apps/content-script/src/app/app.component.ts

```diff
@@ -27,10 +30,63 @@ export class AppComponent implements AfterViewInit {
 
   constructor(
     private bgSub: BackgroundSubService,
-    private logger: LoggerService
+    private logger: LoggerService,
+    private fileService: PlopdownFileService,
   ) {
-    this.videoRefs$ = this.bgSub.getVideoRefsFound().pipe(
+
+    const linkVideoRefs = new Observable(subscriber => {
+      if (window.location.hash.startsWith("#plopdown:")) {
+        const linkedVideo = LZString.decompressFromEncodedURIComponent(window.location.hash.split(":")[1]);
+        subscriber.next(linkedVideo);
+      }
+      subscriber.complete();
+    })
+
+    function convertVideoRef(file: PlopdownFile): [SavedVideoRef] {
+      const s: SavedTrack = {
+        _id: "local",
+        _rev: "local",
+        title: file.headers.title,
+        for: file.headers.for,
+        created: file.headers.created,
+        thumbnail: file.headers.thumbnail,
+        authors: file.headers.authors,
+        language: file.headers.language,
+        license: file.headers.license,
+        cues: file.cues,
+      };
+
+      const v: SavedVideoRef = {
+        _id: "local",
+        _rev: "local",
+        xpath: file.headers.xpath,
+        title: file.headers.for,
+        duration: 0,
+        frameOrigin: "",
+        track: s,
+        frameTitle: null,
+        framePath: null,
+        frameSearch: null,
+      }
+      return [v]
+    }
+
+    function importVTT(input: string): PlopdownFile {
+      try {
+        const f: PlopdownFile = fileService.decode(input);
+        return f
+      } catch(e) {
+        console.error(e);
```

Comment:
Use the logger service here instead of console.error (it'll store the error locally for debugging later)